src/FirebaseInterface.ts

```diff
@@ -31,14 +31,14 @@ export class FirebaseInterface {
     public static tableName = '';
 
     /// Data loaded from FB
-    public data: {};
+    public data: T;
 
     public loginDataInstance: () => LoginData;
 
         /// Keys to load from FB
-    public keys: {}[];
+    public keys: { key: string }[];
```

Comment:
i’m not sure this has to be a string, and may have more information about the key than just the key itself

Reply:
what other information would it need?

Reply:
That would be dependant on the project, so I think we should leave this generic and define the structure in a project level subclass

Reply:
You can always assign an object with more properties to an object that needs less properties. e.g. You can do this: `const abc: { key: string } = { key: 'abc', color: 'red' }`

Reply:
How about this: `{ key: string. [key: string]: any }`. This will ask for a minimum of `key` but also allow other properties

Reply:
sure